### DISCUSSION TITLE
auth.uid() deprecation timeline

### DISCUSSION THREAD

**Discussion by whollacsek** [supabase/supabase#15787] [Questions]
`auth.uid()` has been deprecated for some time now but it is still everywhere in the docs. However, replacing it with `(auth.jwt()->>'sub')::uuid` doesn't seem like a good solution. Can the team confirm this will be the way forward or `auth.uid()` will be kept around for longer?

**Comment by j4w8n** [comment]
How is it deprecated? I'm not familiar with it's origin story.

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/auth/row-level-security
No mention of deprecation in the current docs.  I know at one time I saw it mentioned and maybe was even a plan, but I assume that changed.

**Reply by whollacsek** [reply]
If you inspect the function this is the definition:

```sql
CREATE FUNCTION uid() RETURNS uuid
    STABLE
    LANGUAGE sql
AS
$$
  select 
  coalesce(
    nullif(current_setting('request.jwt.claim.sub', true), ''),
    (nullif(current_setting('request.jwt.claims', true), '')::jsonb ->> 'sub')
  )::uuid
$$;

COMMENT ON FUNCTION uid() IS 'Deprecated. Use auth.jwt() -> ''sub'' instead.';

ALTER FUNCTION uid() OWNER TO supabase_auth_admin;

GRANT EXECUTE ON FUNCTION uid() TO dashboard_user;
```

Same deprecation message for `auth.email()` and `auth.role()`

**Reply by GaryAustin1** [reply]
That is where I saw it but I think they realized the impact of doing that was huge. I seem to recall that conversation somewhere. Anyway unless an employee familiar with it comes along I would not be worrying about it happening at least for a long time as it is not documented to be deprecated in the documentation and the other two are.  They have been deprecated for a year with notice for a year at least and still work.

**Reply by GaryAustin1** [reply]
I guess one of us can generate an issue in gotrue repository to clarify the intent and remove the comment here: https://github.com/supabase/gotrue/blob/master/migrations/20220531120530_add_auth_jwt_function.up.sql if not planned.
